TARS/src/mcu_main/telemetry.h

```diff
@@ -25,30 +26,46 @@ struct TelemetryDataLite {
     int16_t highG_ax;             //[128, -128]
     int16_t highG_ay;             //[128, -128]
     int16_t highG_az;             //[128, -128]
-    int16_t gyro_x;               //[-4096, 4096]
-    int16_t gyro_y;               //[-4096, 4096]
-    int16_t gyro_z;               //[-4096, 4096]
+    int16_t bno_roll;              //[-4,4]
+    int16_t bno_pitch;            //[-4,4]
+    int16_t bno_yaw;             //[-4,4]
 
-    uint8_t flap_extension;  //[0, 256]
-    uint8_t barometer_temp;  //[0, 128]
+    float flap_extension;  //[0, 256]
 };
 
 struct TelemetryPacket {
     TelemetryDataLite datapoints[4];
     float gps_lat;
     float gps_long;
     float gps_alt;
+    float yaw;
+    float pitch;
+    float roll;
     float gnc_state_x;
     float gnc_state_vx;
     float gnc_state_ax;
+    float gnc_state_y;
+    float gnc_state_vy;
+    float gnc_state_ay;
+    float gnc_state_z;
+    float gnc_state_vz;
+    float gnc_state_az;
     float gns_state_apo;
+    int16_t mag_x;                //[-4, 4]
+    int16_t mag_y;                //[-4, 4]
+    int16_t mag_z;                //[-4, 4]
+    int16_t gyro_x;               //[-4096, 4096]
+    int16_t gyro_y;               //[-4096, 4096]
+    int16_t gyro_z;               //[-4096, 4096]
     int16_t response_ID;      //[0, 2^16]
     int8_t rssi;              //[-128, 128]
     int8_t datapoint_count;   //[0,4]
     uint8_t voltage_battery;  //[0, 16]
     uint8_t FSM_State;        //[0,256]
+    int16_t barometer_temp;  //[-128, 128]
 };
 
+
 // Commands transmitted from ground station to rocket
 enum CommandType { SET_FREQ, SET_CALLSIGN, ABORT, TEST_FLAPS, EMPTY };
```

Comment:
How many of these are actually implemented?

Reply:
All of them other than TEST_FLAPS have currently been implemented. We had test flaps at one point but removed it from production because it served no purpose in flight. I think we should bring it back tho since it's pretty useful for debugging. Also it would remove the need for software people to be present for length-to-angle conversion stuff since we could just use a flaps command to write a specific angle to the servos.